src/cdk/wizards/previewStateMachineCDKWizard.ts

```diff
@@ -89,50 +102,92 @@ export default class PreviewStateMachineCDKWizard extends MultiStepWizard<Previe
+        const TOP_LEVEL_NODES: TopLevelNodePickItem[] = []
 
-        const STATE_MACHINES: ConstructNodePickItem[] = []
-        const topLevelNodes: ConstructNode[] = []
         constructNodes.map(node => {
-            //if(node.contextValue === 'awsCdkConstructNode' || node.contextValue === 'awsCdkStateMachineNode'){
-            topLevelNodes.push(node as ConstructNode)
-            //}
+            TOP_LEVEL_NODES.push({
+                label: node.label ? node.label : '',
+                topLevelNode: node as ConstructNode
+            })
         })
 
-        //what about nested construct nodes??
-        while(topLevelNodes.length>0){
-            const tester = await topLevelNodes.pop()?.getChildren()
-            if (tester) {
-                tester.map(async node => {
-                    //const tester = await node.getChildren()
-                    if (node.contextValue === 'awsCdkStateMachineNode') {
-                        STATE_MACHINES.push({
-                            label: node.label ? node.label : '',
-                            stateMachineNode: node as ConstructNode
-                        })
-                    }
-                })
-            }
+        if (TOP_LEVEL_NODES.length === 0) {
+            TOP_LEVEL_NODES.push({
+                label: `[No cdk application(s) found in workspace '${getCDKAppWorkspaceName(appLocation.cdkJsonPath)}']`,
+                topLevelNode: undefined
+            })
         }
 
         const quickPick = picker.createQuickPick({
             options: {
                 ignoreFocusOut: true,
                 title: localize(
                     'AWS.message.prompt.selectCDKStateMachine.placeholder',
-                    'Select State Machine'
+                    'Select CDK Application'
                 ),
                 step: 2,
-                totalSteps: 2,
+                totalSteps: 3,
+            },
+            buttons: [vscode.QuickInputButtons.Back],
+            items: TOP_LEVEL_NODES,
+        })
+
+        const choices = await this.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        })
+
+        this.topLevelNode = picker.verifySinglePickerOutput<TopLevelNodePickItem>(choices)
+
+        return this.topLevelNode ? wizardContinue(this.SELECT_STATE_MACHINE_ACTION) : WIZARD_GOBACK
+    }
+
+    private readonly SELECT_STATE_MACHINE_ACTION: WizardStep = async () => {
+        const topLevelNode = this.topLevelNode ? this.topLevelNode : undefined
+        const STATE_MACHINES: ConstructNodePickItem[] = []
+        const topLevelNodes = await topLevelNode?.topLevelNode?.getChildren()
+
+        if (topLevelNodes && topLevelNodes.length > 0) {
+            topLevelNodes.map(async node => {
+                if (node.contextValue === 'awsCdkStateMachineNode') {
```

Comment:
Can use a filter chained to a map state to first filter out all nodes that aren't state machine nodes, then map each of the state machine nodes into PickItem object format.